src/components/MetadataPanel/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useMemo } from "react";
+import { Divider } from "antd";
+import classNames from "classnames";
+
+import { AgentMetadata, PositionRotation } from "../../constants/interfaces";
+import { FilledCaret } from "../Icons";
+import { AgentDisplayNode } from "../AgentTree";
+
+import styles from "./style.css";
+
+const MetadataLabels: Record<keyof AgentMetadata, JSX.Element> = {
+    uniqueId: <span>Unique ID</span>,
+    agentType: <span>Agent Type</span>,
+    position: <span>Position</span>,
+    rotation: (
+        <>
+            Rotation
+            <br />
+            (degrees)
+        </>
+    ),
+    radius: <span>Radius</span>,
+};
+
+const formatFloatForDisplay = (float: number): string => {
+    return parseFloat(float.toPrecision(2)).toString();
+};
+interface MetadataPanelProps {
+    selectedAgent: AgentMetadata;
+    uiDisplayData: AgentDisplayNode[];
+}
+
+const MetadataPanel: React.FC<MetadataPanelProps> = ({
+    selectedAgent,
+    uiDisplayData,
+}) => {
+    const [panelExpanded, setPanelExpanded] = React.useState(false);
+    const [previousId, setPreviousId] = React.useState(0);
+
+    const agentSelected = !!selectedAgent.uniqueId;
+    const selectedAgentNotRendered =
+        selectedAgent.agentType >= uiDisplayData.length;
+
+    useEffect(() => {
+        if (!agentSelected) {
+            setPanelExpanded(false);
+            return;
+        }
+        if (previousId !== selectedAgent.uniqueId) {
+            setPanelExpanded(true);
+        }
+        setPreviousId(selectedAgent.uniqueId);
+    }, [selectedAgent]);
+
+    const getFormattedValue = (
+        key: string,
+        value: number | PositionRotation
+    ): JSX.Element | string => {
+        if (selectedAgentNotRendered) {
+            return "-";
+        }
+        if (key === "agentType") {
+            return uiDisplayData[selectedAgent.agentType].title;
+        }
+        if (typeof value === "object") {
```

Comment:
could this also be `key === [KEY_NAME]`? just to make it more explicit?

Reply:
oh, I see, it's for both rotation and position. I might even switch it to if (key === position or key=== rotation) just to make it clear

Reply:
yeah will do, i tried to do without the key argument but added it back in the last change set, so i'll make this more explicit

Reply:
FYI: this change requires keeping `typeof value === "object"` when adding the key checks, so that I can access the properties `value.x` etc, and I need to typecast the argument in the final return as `value as number`